# Tutorials and Manuals for the Spacehogs Lab

## Welcome!

You've landed on the central repository for tutorials, manuals, and how-to documents for the Spacehogs lab. In some cases, tutorials will take the form of Python scripts and example data for you to learn from and play with (e.g. the multiprocessing with Python tutorial written by Kenyon). In other cases, we'll provide links to Google Docs with manuals and explanations of how to do various things in the lab. Those are linked below. Please feel free to add to this repository; if you're not already part of the Carleton Spacehogs GitHub group and you'd like to be, let me know and I'll add you.

#### 1. Multiprocessing with Python.
See the Python scripts and files in the folder called "multiprocessing-python". Written by Kenyon Nystrom, summer 2021.

#### 2. Creating molecular clocks and reconciling gene and species trees using AnGST.
Google Doc is located here: https://docs.google.com/document/d/1PfI23VMWGelkVY8S8JYWc3Utk3twMy82G12a8MRtsc0/edit?usp=sharing
Written by Katherine Mateos, summer 2020.

#### 3. Reconciling gene and species trees using ecceTERA.
Google Doc is located here: https://docs.google.com/document/d/1bVdBwOAgYFkVzXC1qXo_bbUPEPjbxliiFd2j9L5HoK4/edit?usp=sharing
Written by Garrett Chappell, summer 2020.
Here is a slightly more updated Google Doc, based on the methods above: https://docs.google.com/document/d/1Cu9rDoOFIhRMGvLSCIuH4u0AxuEm5968JTbwl8KCwe4/edit#heading=h.3ie8tf8eheeu
Written by Aya Klos, summer 2023.

#### 4. Using machine learning (random forest) to identify genes enriched in specific metagenomes; batch downloading metagenomes from JGI.
Google Doc is located here: https://docs.google.com/document/d/1NU0MKZ7zvMargGlsG9O1Y6a5V5nMPLtxbw2pXOC5XYc/edit?usp=sharing
Written by Qianzi Li, spring 2021.

#### 5. Shared Google Drive for docs, files, photos, etc.
https://drive.google.com/drive/folders/129ttgkHowS5TwmbjXelmxw9gxmmvAFit?usp=sharing

#### 6. General bioinformatics methods from the Genomics and Bionformatics course taught at Carleton College
Read the Docs link here: https://carleton-bioinformatics.readthedocs.io/en/latest/index.html

#### 7. If you want to use RStudio on the server, point a web browser to this URL:
https://baross.its.carleton.edu/rstudio/
It’s only visible from the Carleton campus, or when logged on to the Carleton VPN. It’s good practice to make sure you’re on the Carleton VPN regardless whenever you’re logging on to baross from off campus.
The commercial version of rstudio server sets a memory limit to 1 GB, and we’ve set the memory limit to 6GB. 

